core/src/main/scala/org/bitcoins/core/protocol/Address.scala

```diff
@@ -66,11 +65,11 @@ sealed abstract class P2SHAddress extends BitcoinAddress {
   */
 sealed abstract class Bech32Address extends BitcoinAddress {
 
-  def hrp: HumanReadablePart
+  def hrp: BtcHumanReadablePart
```

Comment:
This should stay as a `HumanReadablePart` -- and we should create a 

```scala

sealed abstract class BtcBech32Address extends Bech32Address { 
  override def hrp: BtcHumanReadablePart
  override def networkParameters: BitcoinNetwork 
  //etc 
}
```

Reply:
This would require `Bech32Address` to extend something else though, and a general rework of how the different addresses work. All the other addresses extend `BitcoinAddress`, after all. Perhaps this would be better to do at a later point, when more coins are added to Bitcoin-S?